src/fields2cover/utils/parser.cpp

```diff
@@ -103,46 +105,97 @@ F2CCell getCellFromJson(const json& imported_cell) {
   return cell;
 }
 
+std::string gen_random(const int len) {
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+  std::string tmp_s;
+  tmp_s.reserve(len);
 
-int Parser::importJson(const std::string& file, F2CFields& fields) {
-  std::ifstream f(file);
-  json imported_field = json::parse(f);
+  for (int i = 0; i < len; ++i) {
+    tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+  }
 
+  return tmp_s;
+}
+
+int importJsonData(json imported_field, F2CFields& fields) {
   for (auto&& imported_cell : imported_field["features"]) {
+    auto name = imported_cell["properties"].contains("Name")
+                    ? imported_cell["properties"]["Name"].get<std::string>()
+                    : "cell " + gen_random(3);
     fields.emplace_back(
-      F2CField(F2CCells(getCellFromJson(imported_cell)),
-        imported_cell["properties"]["Name"]));
+        F2CField(F2CCells(getCellFromJson(imported_cell)), name));
   }
   return 0;
 }
 
+int Parser::importJson(const std::string& file, F2CFields& fields) {
+  std::ifstream f(file);
+  json imported_field = json::parse(f);
+
+  return importJsonData(imported_field, fields);
+}
+
+int Parser::importJsonFromString(const std::string& data, F2CFields& fields) {
+  json imported_field = json::parse(data);
+
+  return importJsonData(imported_field, fields);
+}
+
 F2CCell Parser::importCellJson(const std::string& file) {
   std::ifstream f(file);
   json imported_field = json::parse(f);
   return getCellFromJson(imported_field["features"][0]);
 }
 
-F2CSwaths Parser::importSwathsJson(const std::string& file) {
-  std::ifstream f(file);
-  json imported_swaths = json::parse(f);
+F2CCell Parser::importCellJsonFromString(const std::string& jsonString) {
+  json imported_field = json::parse(jsonString);
+  return getCellFromJson(imported_field["features"][0]);
+}
+
+F2CSwaths importSwathsJsonData(const json& data) {
   F2CSwaths swaths;
-  for (auto&& imported_swath : imported_swaths["features"]) {
+  int i = 0;
+  for (auto&& imported_swath : data["features"]) {
     F2CLineString line;
     for (auto&& ps : imported_swath["geometry"]["coordinates"]) {
       line.addPoint(getPointFromJson(ps));
     }
-    swaths.emplace_back(line,
-      imported_swath["properties"]["width"],
-      imported_swath["properties"]["path_id"]);
+    if (!imported_swath["properties"].contains("width")) {
+      std::cout << "Warning: Width property is missing for imported swath, "
+                   "defaulting to 0.1"
+                << std::endl;
+    }
+
+    auto width = imported_swath["properties"].contains("width")
+                     ? imported_swath["properties"]["width"].get<double>()
+                     : 0.1;  // do not fail if the width is not present
```

Comment:
This should be a flag. What if the user want to make sure the width is imported and something is failing?

Reply:
Here I mean a different thing with flag as in the previous comment. Here I would say to add an argument to the function called `fail_without_swath_width`, or something like this, and it should have a default value of false.